### PR TITLE
Use pytest

### PR DESCRIPTION
**Pull request by toslunar**
- Change `nose` to `pytest`
- Move base classes `_Test*` of tests to files `basetest_*.py`, because `pytest` cannot ignore classes by their names, but by the filenames.

### CODE REVIEW

**Comment by muupan** [comment]
Why did coverage increase that much?  > Coverage increased (+6.5%) to 79.691%

**Comment by toslunar** [comment]
It seems nosetests didn't report the coverage by subprocess.
I'm not sure if there's other reason why the coverage increased.

File: tests/q_functions_tests/basetest_state_action_q_function.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    *testing.product({
+        'n_dim_obs': [1, 5],
+        'n_dim_action': [1, 3],
+        'n_hidden_layers': [0, 1, 2],
+        'n_hidden_channels': [1, 2],
+        'nonlinearity': ['relu', 'elu'],
+        'last_wscale': [1, 1e-3],
+    })
+)
+class TestFCSAQFunction(_TestSAQFunction):
```
Comment (muupan) [inline tests/q_functions_tests/basetest_state_action_q_function.py]: This is a concrete test case, not a base test. It looks the same as `TestFCSAQFunction` in `tests/q_functions_tests/test_state_action_q_function.py`, so I think it is not necessary.

File: tests/q_functions_tests/basetest_state_action_q_function.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    *testing.product({
+        'n_dim_obs': [1, 5],
+        'n_dim_action': [1, 3],
+        'n_hidden_layers': [0, 1, 2],
+        'n_hidden_channels': [1, 2],
+        'nonlinearity': ['relu', 'elu'],
+        'last_wscale': [1, 1e-3],
+    })
+)
+class TestFCLSTMSAQFunction(_TestSAQFunction):
```
Comment (muupan) [inline tests/q_functions_tests/basetest_state_action_q_function.py]: Ditto.

File: tests/q_functions_tests/basetest_state_action_q_function.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        'n_dim_obs': [1, 5],
+        'n_dim_action': [1, 3],
+        'n_hidden_layers': [0, 1, 2],
+        'n_hidden_channels': [1, 2],
+        'normalize_input': [True, False],
+        'nonlinearity': ['relu', 'elu'],
+        'last_wscale': [1, 1e-3],
+    })
+)
+class TestFCBNSAQFunction(_TestSAQFunction):
```
Comment (muupan) [inline tests/q_functions_tests/basetest_state_action_q_function.py]: Ditto.

File: tests/q_functions_tests/basetest_state_action_q_function.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        'n_dim_obs': [1, 5],
+        'n_dim_action': [1, 3],
+        'n_hidden_layers': [1, 2],  # LateAction requires n_hidden_layers >=1
+        'n_hidden_channels': [1, 2],
+        'normalize_input': [True, False],
+        'nonlinearity': ['relu', 'elu'],
+        'last_wscale': [1, 1e-3],
+    })
+)
+class TestFCBNLateActionSAQFunction(_TestSAQFunction):
```
Comment (muupan) [inline tests/q_functions_tests/basetest_state_action_q_function.py]: Ditto.

**Comment by toslunar** [comment]
Thanks.  Fixed.

**Comment by muupan** [comment]
LGTM

**Pull request merged**